mtags/src/main/scala/scala/meta/internal/mtags/JavaToplevelMtags.scala

```diff
@@ -170,6 +170,8 @@ class JavaToplevelMtags(
       reader.nextChar()
       reader.ch match {
         case `quote` => Token.Literal
+        case _ if reader.endCharOffset >= reader.buf.length =>
```

Comment:
```suggestion
        case Token.EOF =>
```

Reply:
That will not work, since it's a char not a token yet unfortunately.